### ISSUE TITLE
Add native TypeScript definitions

### ISSUE THREAD

**Issue by puppy0cam** [jezen/is-thirteen#630]
my code experiences many runtime errors due to improper implementation of this module. To resolve this, I recommend that type definitions are created so that the IDEs that support TypeScript are able to provide code suggestions.

**Comment by Almenon** [comment]
I totally agree with this - typescript is awesome! I just migrated [python-shell](https://github.com/extrabacon/python-shell) over to typescript.
